### DISCUSSION TITLE
Silent option not working commented out line 65 in  io.py

### DISCUSSION THREAD

**Discussion by jrussellwhite** [chezou/tabula-py#363] [Q&A] [answered]
# Summary of your issue

I have a PDF file that was throwing tons of warnings. I tried everything to enforce the silent option per the documentation. Finally, I went into my local copy of io.py and commented out line 65 (if options.get("silent"): ) and I was able to suppress the superfluous warnings. Either the code is wrong, or the documentation is inaccurate. I'm not sure which it is. If it was my usage of tabula.read_pdf  please update the documentation to specifically provide an example of how to silence java warnings.

# Check list before submit

- [ X] Did you read [FAQ](https://tabula-py.readthedocs.io/en/latest/faq.html)?

- [ ] (Optional, but really helpful) Your PDF URL: ?

- [X ] Paste the output of `import tabula; tabula.environment_info()` on Python REPL: ?
Python version:
    3.9.2 (tags/v3.9.2:1a79785, Feb 19 2021, 13:44:55) [MSC v.1928 64 bit (AMD64)]
Java version:
    java version "1.8.0_291"
Java(TM) SE Runtime Environment (build 1.8.0_291-b10)
Java HotSpot(TM) Client VM (build 25.291-b10, mixed mode, sharing)
tabula-py version: 2.2.0
platform: Windows-10-10.0.19041-SP0
uname:
    uname_result(system='Windows', node='xxxx', release='10', version='10.0.19041', machine='AMD64')
linux_distribution: ('', '', '')
mac_ver: ('', ('', '', ''), '')

If not possible to execute `tabula.environment_info()`, please answer following questions manually.

- [ ] Paste the output of `python --version` command on your terminal: ?
- [ ] Paste the output of `java -version` command on your terminal: ?
- [ ] Does `java -h` command work well?; Ensure your java command is included in `PATH`
- [ ] Write your OS and it's version: ?

# What did you do when you faced the problem?

## Code:

 tables = tabula.read_pdf(pdfDocument, pages="all", lattice=True)

paste your core code which minimum reproducible for the issue
```

## Expected behavior:

dataframes returned with no java warnings

```
write your expected output
dataframes returned with no java warnings

## Actual behavior:
dataframes returned with the following warnings

Got stderr: May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDSimpleFont toUnicode
WARNING: No Unicode mapping for g1 (2) in font PHMPLI+ArialMT     
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+111 (111) in font GEInspiraSans-Regular-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+3 (3) in font GEInspiraSans-Regular-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+77 (77) in font GEInspiraSans-Regular-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+106 (106) in font GEInspiraSans-Regular-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+122 (122) in font GEInspiraSans-Regular-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+117 (117) in font GEInspiraSans-Regular-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+116 (116) in font GEInspiraSans-Regular-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+103 (103) in font GEInspiraSans-Regular-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+114 (114) in font GEInspiraSans-Regular-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+109 (109) in font GEInspiraSans-Regular-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+107 (107) in font GEInspiraSans-Regular-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+120 (120) in font GEInspiraSans-Regular-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+108 (108) in font GEInspiraSans-Regular-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+65 (65) in font GEInspiraSans-Regular-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+31 (31) in font GEInspiraSans-Regular-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+66 (66) in font GEInspiraSans-Regular-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+64 (64) in font GEInspiraSans-Regular-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+115 (115) in font GEInspiraSans-Regular-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+89 (89) in font GEInspiraSans-Regular-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+99 (99) in font GEInspiraSans-Regular-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+110 (110) in font GEInspiraSans-Regular-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+123 (123) in font GEInspiraSans-Regular-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+127 (127) in font GEInspiraSans-Regular-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+104 (104) in font GEInspiraSans-Regular-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+121 (121) in font GEInspiraSans-Regular-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+60 (60) in font GEInspiraSans-Regular-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+58 (58) in font GEInspiraSans-Regular-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+105 (105) in font GEInspiraSans-Regular-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+118 (118) in font GEInspiraSans-Regular-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+84 (84) in font GEInspiraSans-Regular-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+94 (94) in font GEInspiraSans-Regular-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+61 (61) in font GEInspiraSans-Bold-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+3 (3) in font GEInspiraSans-Bold-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+3 (3) in font GEInspiraSans-Italic-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+103 (103) in font GEInspiraSans-Italic-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+116 (116) in font GEInspiraSans-Italic-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+106 (106) in font GEInspiraSans-Italic-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.graphics.color.PDICCBased ensureDisplayProfile
WARNING: ICC profile is Perceptual, ignoring, treating as Display 
class
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+1 (1) in font GEInspiraSans-Italic-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+1 (1) in font GEInspiraSans-BoldItalic-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+106 (106) in font GEInspiraSans-Italic-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+101 (101) in font GEInspiraSans-Italic-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+102 (102) in font GEInspiraSans-Italic-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+118 (118) in font GEInspiraSans-Italic-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+109 (109) in font GEInspiraSans-Italic-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+103 (103) in font GEInspiraSans-Italic-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+120 (120) in font GEInspiraSans-Italic-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+115 (115) in font GEInspiraSans-Italic-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+114 (114) in font GEInspiraSans-Italic-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+94 (94) in font GEInspiraSans-Bold-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+101 (101) in font GEInspiraSans-Bold-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unicode mapping for CID+119 (119) in font GEInspiraSans-Bold-Identity-H
May 07, 2021 4:47:21 PM org.apache.pdfbox.pdmodel.font.PDType0Font toUnicode
WARNING: No Unico

**Comment by chezou** [comment, marked as answer]
@jrussellwhite I think using `silent` option should suppress the warning. Can you try it?

```py
tabula.read_pdf(pdfDocument, pages="all", lattice=True, silent=True)
```

Here is the document for that (see silent option section): https://tabula-py.readthedocs.io/en/latest/tabula.html#tabula.io.build_options

At least, it looks like you didn't use `silent` option. Could you share me exact code you wrote?

**Comment by jrussellwhite** [comment]
Darn do I feel stupid. I swear I thought that was the first thing I tried. That does work. Thanks for the quick response. Sorry for bothering you. 

(Sent from my phone. Typos courtesy of iOS)

**Comment by chezou** [comment]
Nvm! Glad to hear that.

**Comment by wasiongit** [comment]
It really helps thanks @chezou and @jrussellwhite.
